src/main/java/com/icthh/xm/ms/entity/web/rest/FunctionResource.java

```diff
@@ -184,4 +184,25 @@ public static String getFunctionKey(HttpServletRequest request) {
         String bestMatchingPattern = request.getAttribute(BEST_MATCHING_PATTERN_ATTRIBUTE).toString();
         return new AntPathMatcher().extractPathWithinPattern(bestMatchingPattern, path);
     }
+
+    private ResponseEntity<Object> processCreatedResponce(FunctionContext result) {
+        URI uri = buildCtxUri(result);
+        if (uri == null) {
+            return ResponseEntity
+                .status(HttpStatus.CREATED.value())
+                .body(result.functionResult());
+        } else {
+            return ResponseEntity.created(uri)
+                .headers(HeaderUtil.createEntityCreationAlert(ENTITY_NAME_FUNCTION_CONTEXT, String.valueOf(result.getId())))
+                .body(result.functionResult());
+        }
+    }
+
+    private URI buildCtxUri(FunctionContext result) {
```

Comment:
Change to Optional

Reply:
it's better to clean code, and will be illuminated by scalar replacement

Reply:
Fixed.
But with Optioanal there are tons of speculation
https://github.com/commonquail/java-optional-or-else-benchmark
https://pkolaczk.github.io/overhead-of-optional/
Read for future discussions. There are lots if possible overhead, and plain old null check is not something dirty.